kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDefinitionController.java

```diff
@@ -155,30 +148,34 @@ private void initialize() {
         doneButton.disableProperty().bind(patternDefinitionViewModel.getProperty(IS_INVALID));
 
         setupDragNDrop(purposeStackPane, (publicId) -> {
-            // check to see if a pattern > purpose was already dragged into the purpose section before saving
-            // to the view model
-            if (patternDefinitionViewModel.getPropertyValue(PURPOSE_ENTITY) == null) {
-                // query public Id to get entity.
-                Entity entity = EntityService.get().getEntityFast(EntityService.get().nidForPublicId(publicId));
-                patternDefinitionViewModel.setPropertyValue(PURPOSE_ENTITY, entity);
-                addPurposeToForm(entity);
-            }
+            // query public Id to get entity.
+            Entity entity = EntityService.get().getEntityFast(EntityService.get().nidForPublicId(publicId));
+            patternDefinitionViewModel.setPropertyValue(PURPOSE_ENTITY, entity);
         });
 
         setupDragNDrop(meaningStackPane, (publicId) -> {
-            // check to see if a pattern > purpose was already dragged into the purpose section before saving
-            // to the view model
-            if (patternDefinitionViewModel.getPropertyValue(MEANING_ENTITY) == null) {
-                // query public Id to get entity.
-                Entity entity = EntityService.get().getEntityFast(EntityService.get().nidForPublicId(publicId));
-                patternDefinitionViewModel.setPropertyValue(MEANING_ENTITY, entity);
-                addMeaningToForm(entity);
-            }
+            // query public Id to get entity.
+            Entity entity = EntityService.get().getEntityFast(EntityService.get().nidForPublicId(publicId));
+            patternDefinitionViewModel.setPropertyValue(MEANING_ENTITY, entity);
         });
-        ObjectProperty<ConceptEntity> purposeProp = patternDefinitionViewModel.getProperty(PatternFieldsViewModel.PURPOSE_ENTITY);
-        ObjectProperty<ConceptEntity> meaningProp = patternDefinitionViewModel.getProperty(PatternFieldsViewModel.MEANING_ENTITY);
+        ObjectProperty<ConceptEntity> purposeProp = patternDefinitionViewModel.getProperty(PURPOSE_ENTITY);
+        ObjectProperty<ConceptEntity> meaningProp = patternDefinitionViewModel.getProperty(MEANING_ENTITY);
         purposeProp.addListener(fieldsValidationListener);
+        purposeProp.addListener((obs, oldVal, newVal) -> {
+            if(newVal !=null){
+                addPurposeToForm(newVal);
+            }else{
+                removePurposeForm();
+            }
+        });
```

Comment:
if we're not using the old value perhaps an InvalidationListener would do? (slightly better performance)